### ISSUE TITLE
Bug: Helper function directory_mirror() throws an error if destination directory exists (with overwrite option true)

### ISSUE THREAD

**Issue by dmillard70** [codeigniter4/CodeIgniter4#5478]
### PHP Version

8.0

### CodeIgniter4 Version

4.1.5

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### Database

MySql 5.6

### What happened?

Helper function directory_mirror() throws an error if destination directory exists (with overwrite option true), because there is no check whether directory already exists (filesystem_helper.php line 94).
I think this should be changed because the name of the function (* mirror *) suggests a function including subdirectories.

### Steps to Reproduce

Just call twice directory_mirror() with same source (including subdirectories) and target path with overwrite option true.

### Expected Output

Create sub directory only if subdirectory in target does not exist. 
Consequently the name (* mirror *) perhaps suggests that you could mirror an path with all elements (including deleting no longer existing elements in destination)

**Comment by kenjis** [comment]
I confirmed the bug.

```php
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function index()
    {
        helper('filesystem');
        directory_mirror(APPPATH . 'Config', WRITEPATH . 'tmp/Config');
    }
}
```
```sh-session
$ php public/index.php
$ php public/index.php 


[ErrorException]

mkdir(): File exists

at SYSTEMPATH/Helpers/filesystem_helper.php:94

Backtrace:
  1    [internal function]
       CodeIgniter\Debug\Exceptions()->errorHandler(2, 'mkdir(): File exists', '/Users/kenji/work/codeigniter/CodeIgniter4/system/Helpers/filesystem_helper.php', 94)

  2    SYSTEMPATH/Helpers/filesystem_helper.php:94
       mkdir('/Users/kenji/work/codeigniter/CodeIgniter4/writable/tmp/Config/Boot', 493)

  3    APPPATH/Controllers/Home.php:10
       directory_mirror('/Users/kenji/work/codeigniter/CodeIgniter4/app/Config', '/Users/kenji/work/codeigniter/CodeIgniter4/writable/tmp/Config')

  4    SYSTEMPATH/CodeIgniter.php:823
       App\Controllers\Home()->index()

  5    SYSTEMPATH/CodeIgniter.php:410
       CodeIgniter\CodeIgniter()->runController(Object(App\Controllers\Home))

  6    SYSTEMPATH/CodeIgniter.php:318
       CodeIgniter\CodeIgniter()->handleRequest(null, Object(Config\Cache), false)

  7    FCPATH/index.php:37
       CodeIgniter\CodeIgniter()->run()
```

**Comment by kenjis** [comment]
@dmillard70 I sent a PR to fix: #5493
